src/System.Memory/src/System/Buffers/ReadOnlySequence.Helpers.cs

```diff
@@ -102,44 +102,51 @@ private ReadOnlyMemory<T> GetFirstBuffer()
             // Highest bit of startIndex: A = startIndex >> 31
             // Highest bit of endIndex: B = endIndex >> 31
 
-            if (startIndex >= 0)
+            // A == 0 && B == 0 means SequenceType.MultiSegment
+            // Equivalent to startIndex >= 0 && endIndex >= 0
+            if ((startIndex >> 31) + (endIndex >> 31) == 0)
```

Comment:
If you want to play cryptic bit tricks, this can be written as `(startIndex | endIndex) >= 0` to save a few more instructions.